File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/element/path/Image.java
package com.tickaroo.tikxml.annotationprocessing.element.path;

import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml
public class Image {

  @Override public boolean equals(Object obj) {
    return obj instanceof Image;
  }
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/element/path/Image.kt
package com.tickaroo.tikxml.annotationprocessing.element.path

import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml
class Image {
    override fun equals(other: Any?): Boolean {
        return other is Image
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}
